etcdserver/api/v3rpc/watch.go

```diff
@@ -67,6 +71,13 @@ func (ws *watchServer) Watch(stream pb.Watch_WatchServer) error {
 }
 
 func (sws *serverWatchStream) recvLoop() error {
+	rh := &pb.ResponseHeader{
```

Comment:
should this get updated inside the `for` loop? otherwise rev and raftterm will be stale